cumulus/pallets/parachain-system/src/validate_block/implementation.rs

```diff
@@ -168,6 +168,11 @@ where
 	let cache_provider = trie_cache::CacheProvider::new();
 	let seen_nodes = SeenNodes::<HashingFor<B>>::default();
 
+	blocks.iter().fold(parent_header.hash(), |p, b| {
```

Comment:
Why not move this even earlier to bail out faster?  Looks like we can do it right after we process `block_data`.